modules/imgproc/src/shapedescr.cpp

```diff
@@ -454,6 +455,369 @@ cv::RotatedRect cv::fitEllipse( InputArray _points )
+                 DMd[1*6+3]*DMd[4*6+5]*DMd[5*6+3] - DMd[1*6+4]*DMd[3*6+3]*DMd[5*6+5] + DMd[1*6+3]*DMd[4*6+3]*DMd[5*6+5];
+    TMd[1*3+2] = DMd[2*6+5]*DMd[3*6+3]*DMd[4*6+5] - DMd[2*6+5]*DMd[3*6+5]*DMd[4*6+3] + DMd[2*6+4]*DMd[3*6+5]*DMd[5*6+3] - \
+                 DMd[2*6+3]*DMd[4*6+5]*DMd[5*6+3] - DMd[2*6+4]*DMd[3*6+3]*DMd[5*6+5] + DMd[2*6+3]*DMd[4*6+3]*DMd[5*6+5];
+    TMd[2*3+0] = DMd[0*6+5]*DMd[3*6+4]*DMd[4*6+3] - DMd[0*6+5]*DMd[3*6+3]*DMd[4*6+4] - DMd[0*6+4]*DMd[3*6+4]*DMd[5*6+3] + \
+                 DMd[0*6+3]*DMd[4*6+4]*DMd[5*6+3] + DMd[0*6+4]*DMd[3*6+3]*DMd[5*6+4] - DMd[0*6+3]*DMd[4*6+3]*DMd[5*6+4];
+    TMd[2*3+1] = DMd[1*6+5]*DMd[3*6+4]*DMd[4*6+3] - DMd[1*6+5]*DMd[3*6+3]*DMd[4*6+4] - DMd[1*6+4]*DMd[3*6+4]*DMd[5*6+3] + \
+                 DMd[1*6+3]*DMd[4*6+4]*DMd[5*6+3] + DMd[1*6+4]*DMd[3*6+3]*DMd[5*6+4] - DMd[1*6+3]*DMd[4*6+3]*DMd[5*6+4];
+    TMd[2*3+2] = DMd[2*6+5]*DMd[3*6+4]*DMd[4*6+3] - DMd[2*6+5]*DMd[3*6+3]*DMd[4*6+4] - DMd[2*6+4]*DMd[3*6+4]*DMd[5*6+3] + \
+                 DMd[2*6+3]*DMd[4*6+4]*DMd[5*6+3] + DMd[2*6+4]*DMd[3*6+3]*DMd[5*6+4] - DMd[2*6+3]*DMd[4*6+3]*DMd[5*6+4];
+
+    Ts=(-(DMd[3*6+5]*DMd[4*6+4]*DMd[5*6+3]) + DMd[3*6+4]*DMd[4*6+5]*DMd[5*6+3] + DMd[3*6+5]*DMd[4*6+3]*DMd[5*6+4] -
+          DMd[3*6+3]*DMd[4*6+5]*DMd[5*6+4]  - DMd[3*6+4]*DMd[4*6+3]*DMd[5*6+5] + DMd[3*6+3]*DMd[4*6+4]*DMd[5*6+5]);
+
+    Md[0*3+0] = (DMd[2*6+0] + (DMd[2*6+3]*TMd[0*3+0] + DMd[2*6+4]*TMd[1*3+0] + DMd[2*6+5]*TMd[2*3+0])/Ts)/2.;
+    Md[0*3+1] = (DMd[2*6+1] + (DMd[2*6+3]*TMd[0*3+1] + DMd[2*6+4]*TMd[1*3+1] + DMd[2*6+5]*TMd[2*3+1])/Ts)/2.;
+    Md[0*3+2] = (DMd[2*6+2] + (DMd[2*6+3]*TMd[0*3+2] + DMd[2*6+4]*TMd[1*3+2] + DMd[2*6+5]*TMd[2*3+2])/Ts)/2.;
+    Md[1*3+0] = -DMd[1*6+0] - (DMd[1*6+3]*TMd[0*3+0] + DMd[1*6+4]*TMd[1*3+0] + DMd[1*6+5]*TMd[2*3+0])/Ts;
+    Md[1*3+1] = -DMd[1*6+1] - (DMd[1*6+3]*TMd[0*3+1] + DMd[1*6+4]*TMd[1*3+1] + DMd[1*6+5]*TMd[2*3+1])/Ts;
+    Md[1*3+2] = -DMd[1*6+2] - (DMd[1*6+3]*TMd[0*3+2] + DMd[1*6+4]*TMd[1*3+2] + DMd[1*6+5]*TMd[2*3+2])/Ts;
+    Md[2*3+0] = (DMd[0*6+0] + (DMd[0*6+3]*TMd[0*3+0] + DMd[0*6+4]*TMd[1*3+0] + DMd[0*6+5]*TMd[2*3+0])/Ts)/2.;
+    Md[2*3+1] = (DMd[0*6+1] + (DMd[0*6+3]*TMd[0*3+1] + DMd[0*6+4]*TMd[1*3+1] + DMd[0*6+5]*TMd[2*3+1])/Ts)/2.;
+    Md[2*3+2] = (DMd[0*6+2] + (DMd[0*6+3]*TMd[0*3+2] + DMd[0*6+4]*TMd[1*3+2] + DMd[0*6+5]*TMd[2*3+2])/Ts)/2.;
+
+    if (fabs(cv::determinant(M)) > 1.0e-10) {
+        Mat eVal, eVec;
+        EigenvalueDecomposition::eigensystem(M, eVal, eVec);
+
+    // Select the eigen vector {a,b,c} which satisfies 4ac-b^2 > 0
+    double cond[3];
+    cond[0]=(4.0 * eVec.at<double>(0,0) * eVec.at<double>(2,0) - eVec.at<double>(1,0) * eVec.at<double>(1,0));
+    cond[1]=(4.0 * eVec.at<double>(0,1) * eVec.at<double>(2,1) - eVec.at<double>(1,1) * eVec.at<double>(1,1));
+    cond[2]=(4.0 * eVec.at<double>(0,2) * eVec.at<double>(2,2) - eVec.at<double>(1,2) * eVec.at<double>(1,2));
+    if (cond[0]<cond[1]) {
+        (cond[1]<cond[2]) ? i=2 : i=1;
```

Comment:
Code static analyzers dislike assignments in expressions. Please use this form instead:
```
i = (cond[1]<cond[2]) ? 2 : 1;
```